packages/graphql/lib/schema-builder/collections/metadata-by-target.collection.ts

```diff
@@ -1,10 +1,11 @@
 import { MetadataCollectionModel } from './metada-collection-model.interface';
 import { TargetMetadataCollection } from './target-metadata.collection';
+import { ClassMetadata } from '../metadata';
 
 export class MetadataByTargetCollection {
   public readonly all: MetadataCollectionModel = {
     argumentType: [],
-    interface: [],
+    interface: new Map<Function, ClassMetadata>(),
     inputType: [],
     objectType: [],
     resolver: [],
```

Comment:
Do you think we should refactor other collections too? (especially objectType?)

Reply:
I agree, but I'm afraid of introducing a lot of changes. Some tests rely on a specific order (arguments for example), moving it to map would break the order.
I think that maybe there's a reason to rethink how the metadata storage works? Maybe a complete overhaul to have it much more optimized?